### ISSUE TITLE
Chinese characters cannot be displayed

### ISSUE THREAD

**Issue by jiangyi0923** [blish-hud/Blish-HUD#282]
When I built the Chinese translation file, the Chinese characters did not appear when I opened the program
And some places are still English characters
![20200716214805](https://user-images.githubusercontent.com/40364405/87678973-2eb3e580-c7ae-11ea-9016-ae2ec3cfc4e0.png)
![20200716214823](https://user-images.githubusercontent.com/40364405/87678988-34a9c680-c7ae-11ea-9e2d-5b27a6ac477e.png)
![20200716214848](https://user-images.githubusercontent.com/40364405/87678997-370c2080-c7ae-11ea-8779-8dacfef4ee47.png)

**Comment by jiangyi0923** [comment]
[zh.zip](https://github.com/blish-hud/Blish-HUD/files/4931858/zh.zip)
This is the Chinese Blish hudd. resources.DLL

**Comment by dlamkins** [comment]
Thank you @jiangyi0923 I believe we had somebody recently recommend a specific font to be used.

Do you by chance use the Chinese client as well or are you playing outside of China?  If you are using the Chinese client, could you let me know if the game itself uses perhaps a different font to display characters?

**Comment by jiangyi0923** [comment]
Yes, I am a server player in China. We usually use the font of arcDPS plug-in. When I try to build blish-hud and make it Chinese, even if "\Content\ Common \ Menomonion.ttf "font file is replaced, The UI still cannot display Chinese characters.

I even tried to modify the "CharacterRegions" parameter of the "Content\ Common \ Menomonia. Spritefont "file and rebuilt it, but still couldn't display Chinese characters

This is the font file:
[arcdps_font.zip](https://github.com/blish-hud/Blish-HUD/files/4941532/arcdps_font.zip)

**Comment by dlamkins** [comment]
Yes, looks like Microsoft YaHei.

The fonts have to get precompiled with a separate application I used to generate the spritefonts which is probably why that didn't work for you.

We'll probably need the content service to automatically swap what font is used and leave the reference to Menomonia if Chinese is detected as the UI language.

Basically have this function detect if it is currently in Chinese, and, if so, swap Menomonia for YaHei's name.  https://github.com/blish-hud/Blish-HUD/blob/f7bab481dd31e131c012e430879abbbacdd45fb0/Blish%20HUD/GameServices/ContentService.cs#L261-L273

**Comment by dlamkins** [comment]
@jiangyi0923 It seems that `Microsoft YaHei` is an incredibly large font (22 MB!).  Is this font typically already installed on systems or is there perhaps a more light weight font with the appropriate character support available?  I understand that it is expected to be quite large due to the number of characters, but ignoring the licensing issues, including the font nearly doubles the application size and also is quite large to load into memory.

**Comment by Eveink** [comment]
> @jiangyi0923 It seems that `Microsoft YaHei` is an incredibly large font (22 MB!). Is this font typically already installed on systems or is there perhaps a more light weight font with the appropriate character support available? I understand that it is expected to be quite large due to the number of characters, but ignoring the licensing issues, including the font nearly doubles the application size and also is quite large to load into memory.

we can use wqy-microhei, it's opensource alternative of ms yahei,even though ms yahei is located at c:\windows\fonts by default.

**Comment by jiangyi0923** [comment]
> > @jiangyi0923 It seems that `Microsoft YaHei` is an incredibly large font (22 MB!). Is this font typically already installed on systems or is there perhaps a more light weight font with the appropriate character support available? I understand that it is expected to be quite large due to the number of characters, but ignoring the licensing issues, including the font nearly doubles the application size and also is quite large to load into memory.
> 
> we can use wqy-microhei, it's opensource alternative of ms yahei,even though ms yahei is located at c:\windows\fonts by default.

It is difficult to determine what characters need to be used. It would be great if you can support it. I can provide you with a Chinese taco path package for testing characters, but if the game map is updated and you use a small font, this will cause duplication. For the problem of building files, my suggestion is to load the Chinese fonts that come with the system if possible, or use C++ and imgui to develop new plug-ins, so you don't need to think about localization.

**Comment by jiangyi0923** [comment]
> @jiangyi0923看来,`Microsoft YaHei`是一个令人难以置信的大字体(22 MB !)。 这是字体通常已经安装在系统或有可能更重量轻和可用适当的字符支持的字体吗? 我知道它将相当大数量的字符,但忽略了许可证问题,包括字体几乎成倍增加应用程序大小也是相当大的加载到内存中。
https://drive.google.com/file/d/1sOdXtHs0wSJFHnh7ZmxfL8MGGvAvhzuv/view?usp=sharing


> > @jiangyi0923看来,`Microsoft YaHei`是一个令人难以置信的大字体(22 MB !)。 这是字体通常已经安装在系统或有可能更重量轻和可用适当的字符支持的字体吗? 我知道它将相当大数量的字符,但忽略了许可证问题,包括字体几乎成倍增加应用程序大小也是相当大的加载到内存中。
> 
> 我们可以使用wqy-microhei,开源替代yahei女士,女士即使yahei位于c: \ windows \默认字体。

https://drive.google.com/file/d/1sOdXtHs0wSJFHnh7ZmxfL8MGGvAvhzuv/view?usp=sharing

**Comment by dlamkins** [comment]
Thank you.  That will be good for testing.  I'll see if I can get it to load from the system because that would be very useful and make this a lot easier.

**Comment by dlamkins** [comment]
Being worked on as part of #423

**Comment by jiangyi0923** [comment]
I am back 
This is the Chinese translation of the RESX file
[Strings.zip](https://github.com/blish-hud/Blish-HUD/files/7774292/Strings.zip)

**Comment by jiangyi0923** [comment]
![20211224013852](https://user-images.githubusercontent.com/40364405/147353635-cd9ca09e-4054-4a47-841c-1e7db53e8d46.png)
and it Support Chinese

**Comment by fd0712** [comment]
> ![20211224013852](https://user-images.githubusercontent.com/40364405/147353635-cd9ca09e-4054-4a47-841c-1e7db53e8d46.png) and it Support Chinese

Please tell me how to translate it to Chinese.
I don't know how to use ur files make the effect.
